dlpy/tests/test_tensorboard.py

```diff
@@ -0,0 +1,482 @@
+#       the CASPROTOCOL environment variable.
+
+import os
+import shutil
+import swat
+import swat.utils.testing as tm
+from swat.cas.table import CASTable
+from swat.cas.results import CASResults
+from dlpy.model import Model, Optimizer
+from dlpy.sequential import Sequential
+from dlpy.tensorboard import TensorBoard
+from dlpy.layers import (InputLayer, Conv2d, Conv1d, Pooling, Dense, OutputLayer,
+                         Recurrent, Keypoints, BN, Res, Concat, Reshape, GlobalAveragePooling1D)
+from dlpy.utils import caslibify, caslibify_context, file_exist_on_server, DLPyError
+from dlpy.applications import Tiny_YoloV2
+from dlpy.splitting import two_way_split
+import unittest
+
+class TestTensorBoard(unittest.TestCase):
+    '''
+    Please locate the images.sashdat file under the datasources to the DLPY_DATA_DIR.
+    '''
+    server_type = None
+    s = None
+    server_sep = '/'
+    data_dir = None
+    data_dir_local = None
+
+    @classmethod
+    def setUpClass(cls):
+        swat.reset_option()
+        swat.options.cas.print_messages = False
+        swat.options.interactive_mode = False
+
+        cls.s = swat.CAS()
+        cls.server_type = tm.get_cas_host_type(cls.s)
+        cls.server_sep = '\\'
+        if cls.server_type.startswith("lin") or cls.server_type.startswith("osx"):
+            cls.server_sep = '/'
+
+        if 'DLPY_DATA_DIR' in os.environ:
+            cls.data_dir = os.environ.get('DLPY_DATA_DIR')
+            if cls.data_dir.endswith(cls.server_sep):
+                cls.data_dir = cls.data_dir[:-1]
+            cls.data_dir += cls.server_sep
+
+        if 'DLPY_DATA_DIR_LOCAL' in os.environ:
+            cls.data_dir_local = os.environ.get('DLPY_DATA_DIR_LOCAL')
+            if cls.data_dir_local.endswith(cls.server_sep):
+                cls.data_dir_local = cls.data_dir_local[:-1]
+            cls.data_dir_local += cls.server_sep
+
+    def test_tensorboard_init_log_dir(self):
+        try:
+            import tensorflow as tf
+        except:
+            unittest.TestCase.skipTest(self, "tensorflow not found in the libraries")
+
+        model1 = Sequential(self.s, model_table='Simple_CNN1')
+        model1.add(InputLayer(3, 224, 224))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Dense(16))
+        model1.add(OutputLayer(act='softmax', n=2))
+
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        caslib, path, tmp_caslib = caslibify(self.s, path=self.data_dir+'images.sashdat', task='load')
+
+        self.s.table.loadtable(caslib=caslib,
+                               casout={'name': 'eee', 'replace': True},
+                               path=path)
+        # Clean up for DNE
+        shutil.rmtree(self.data_dir + '_TB', ignore_errors=True)
```

Comment:
this is on the server side, if the client and server have different os, this might cause problems.